src/cli/address_extractor.rs

```diff
@@ -0,0 +1,29 @@
+use crate::primitives::bitcoin;
+use anyhow::{Context, Result};
+use std::collections::HashSet;
+use std::path::Path;
+
+pub fn read_and_parse_addresses(file_path: impl AsRef<Path>) -> Result<()> {
+    let s = std::fs::read_to_string(file_path).context("Couldn't read file")?;
+    let set: HashSet<bitcoin::Address> = s
+        .split_whitespace()
+        .filter_map(|word| {
+            let word_without_semicolon = if word.ends_with(';') {
```

Comment:
This is not very robust. Anther use case could be a JSON blob that needs to be correctly parsed. Can you think of a more general way? Perhaps a regex that could fuzzy match things that look like addresses and then use the parsing to be sure.